src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Processors/DelayedSamplingProcessor.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    internal class DelayedSamplingProcessor : ITelemetryProcessor
+    {
+        private readonly AdaptiveSamplingTelemetryProcessor _samplingProcessor;
+        private ITelemetryProcessor _next;
+        private bool _isSamplingEnabled = false;        
+
+        public DelayedSamplingProcessor(ITelemetryProcessor next, ApplicationInsightsLoggerOptions options)
+        {
+            _next = next;
+            _samplingProcessor = new AdaptiveSamplingTelemetryProcessor(options.SamplingSettings, null, next);
```

Comment:
Consider extracting construction of `AdaptiveSamplingTelemetryProcessor` and setting of its excluded/included types into a helper method. Can use it here and in `ApplicationInsightsServiceCollectionExtensions`.